### ISSUE TITLE
Add support for reading detection and measurement events files using the Python API

### ISSUE THREAD

**Issue by oscarhiggott** [quantumlib/Stim#220]
I noticed that while there's a [sample_write](https://github.com/quantumlib/Stim/blob/main/doc/python_api_reference_vDev.md#stim.CompiledDetectorSampler.sample_write) method to write samples of detection events to file, I can't find a method to read these file formats using the Python API (unless it's there and I can't find it). The same is true of the measurement sampler [sample_write](https://github.com/quantumlib/Stim/blob/main/doc/python_api_reference_vDev.md#stim.CompiledMeasurementSampler.sample_write) method. It would be convenient to have a sample_read method or equivalent to read these file formats.

**Comment by Strilanc** [comment]
You're picturing a method that takes a file, a format, an expected number of bits per shot, and returns a 2d optionally-bit-packed numpy array?

**Comment by oscarhiggott** [comment]
Yes pretty much, I think it would need to take both the number of detectors and the number of observables, rather than just bits per shot? It's easy enough for the user to write their own parser (e.g. for the dets format at least, I haven't played around with the others yet), but it might be handy if this was already available in the API for all the formats.

**Comment by Strilanc** [comment]
This was implemented in https://github.com/quantumlib/Stim/pull/242 . There are now methods `stim.read_shot_data_file` and `stim.write_shot_data_file`.

**Comment by oscarhiggott** [comment]
Great, thanks!
